docs/howtos/extending/magicgui.md

```diff
@@ -0,0 +1,784 @@
+---
+jupytext:
+  cell_metadata_filter: -all
+  formats: md:myst
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.12
+    jupytext_version: 1.8.2
+kernelspec:
+  display_name: Python 3
+  language: python
+  name: python3
+---
+(creating-widgets)=
+
+# Creating widgets
+
+Widgets are small composable graphical elements that can be added to the `napari` user
+interface. The easiest way to add a widget is by using
+[`magicgui`](https://pyapp-kit.github.io/magicgui/), a python package that assists
+in building widgets. It is a general abstraction layer on GUI toolkit backends (like
+Qt), with an emphasis on mapping python types to widgets. This enables you to easily
+create widgets using annotations.
+If you require more extensibility though, you can create your own widget `class` that
+subclasses [`QtWidgets.QWidget`](https://doc.qt.io/qt-5/qwidget.html).
+
+This document will describe each widget creation method, in increasing order of
+extensibility;
+
+1. creating a widget from a function and
+   [`magicgui`](https://pyapp-kit.github.io/magicgui/) (simplest but least extensible
+   and flexible)
+2. create a widget class that subclasses a
+   [`magicgui` widget class](https://pyapp-kit.github.io/magicgui/widgets/#the-widget-hierarchy)
+3. create a widget class that subclasses
+   [`QtWidgets.QWidget`](https://doc.qt.io/qt-5/qwidget.html) (most extensible but also
+   the most difficult to implement)
+
+For more examples, the [GUI gallery examples](https://napari.org/stable/_tags/gui.html)
+has some widget examples. Additionally,
```

Comment:
```suggestion
For more examples, see the [GUI gallery examples](https://napari.org/stable/_tags/gui.html). Additionally,
```

Reply:
This is awkward because not all of the 'GUI' examples are widget examples. I've added this in parenthesis.